test/EventDBManager.ts

```diff
@@ -0,0 +1,163 @@
+import { Knex, knex as knexBuilder } from "knex";
+import { JSBIDeserializer } from "../src/util/Serializer";
+import { LiquidityEvent } from "./LiquidityEvent";
+import { SwapEvent } from "./SwapEvent";
+import { DateConverter } from "../src/util/DateConverter";
+import { EventType } from "./EventType";
+
+type LiquidityEventRecord = {
+  id: number;
+  type: number;
+  liquidity: string;
+  amount0: string;
+  amount1: string;
+  tick_lower: number;
+  tick_upper: number;
+  block_number: number;
+  transaction_index: number;
+  log_index: number;
+  date: string;
+};
+
+type SwapEventRecord = {
+  id: number;
+  amount0: string;
+  amount1: string;
+  sqrt_price_x96: string;
+  liquidity: string;
+  tick: number;
+  block_number: number;
+  transaction_index: number;
+  log_index: number;
+  date: string;
+};
+
+export class EventDBManager {
+  private knex: Knex;
+  private static _instance: EventDBManager;
+
+  constructor(knex: Knex) {
+    this.knex = knex;
+  }
+
+  public static get instance(): EventDBManager {
+    if (!EventDBManager._instance) {
+      throw new Error("Please build an instance first!");
+    }
+    return EventDBManager._instance;
+  }
+
+  static buildInstance(dbPath: string): Promise<EventDBManager> {
```

Comment:
I don't insist on that. A `SimulatorClient` should have one `DBManager` and one `EventDBManager`. For now `EventDBManager` is for test case use only and we need to figure out the boundary of the simualtor function and refactor the component.

Reply:
I think we can use the more generic dependency injection, if the singleton is needed, then it can be handled on a higher level, we can point out how to handle them in the docs